### PR TITLE
Connectors: Only use secure servers and allow multi location params

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs [hunk 1]
```diff
@@ -478,5 +478,28 @@ public async Task Office365Users_SearchUsers_V2()
...
+        [Fact]
+        public async Task PetStore_MultiServer()
+        {
+            using var testConnector = new LoggingTestServer(@"Swagger\PetStore.json");
+            var apiDoc = testConnector._apiDocument;
+            var config = new PowerFxConfig();
+
+            using var httpClient = new HttpClient(); // testConnector);
+
+            using var client = new PowerPlatformConnectorClient(
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs:491]: >PowerPlatformConnectorClient [](http://example.com/codeflow?start=35&length=28)

Is PetStore a connector?
Can we access this directly as a rest API (httpClient, not PowerPlatformConnectorClient)?

Comment (LucGenetier) [reply src/tests/Microsoft.PowerFx.Connectors.Tests/PowerPlatformConnectorTests.cs:491]: [https://petstore.swagger.io/](https://petstore.swagger.io/)

File: src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs [hunk 2]
```diff
@@ -100,7 +100,15 @@ internal sealed class ServiceFunction : BuiltinFunction
             }
 
             foreach (var requiredParam in requiredParamInfo)
-                _parameterDescriptionMap.Add(requiredParam.TypedName.Name.Value, requiredParam.Description);
+                if (_parameterDescriptionMap.ContainsKey(requiredParam.TypedName.Name.Value))
+                {
+                    // Can happen when a parameters is in Boby and Query at the same time
+                    _parameterDescriptionMap[requiredParam.TypedName.Name.Value] += $";{requiredParam.Description}";
+                }
+                else
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs:108]: ServiceFunction comes from PowerApps, and we're trying to stay mirrored to that. 

How does PowerApps handle this situation today? Ie, should this logic be somewhere else (not changing this function, which will hopefully be shared with PA)

Comment (MikeStall) [reply src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs:108]: alternatively, if we do keep this code:
1. wrap in { } 
2. don't repeat 'requiredParam.TypedName.Name.Value' 3 times. 

            foreach (var requiredParam in requiredParamInfo)
            {
                var key = requiredParam.TypedName.Name.Value;
                if (_parameterDescriptionMap.ContainsKey(key))
                {
                    // Can happen when a parameters is in Body and Query at the same time
                    _parameterDescriptionMap[key] += $";{requiredParam.Description}";
                }
                else
                {
                    _parameterDescriptionMap.Add(key, requiredParam.Description);
                }
            }

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Review by MikeStall — APPROVED** [review]
:shipit:

File: src/tests/Microsoft.PowerFx.Connectors.Tests/BasicRestTests.cs [hunk 3]
```diff
@@ -167,5 +167,19 @@ public void BasicHttpBinding()
...
+        {
+            using var testConnector = new LoggingTestServer(@"Swagger\PetStore.json");
+            var apiDoc = testConnector._apiDocument;
+            var config = new PowerFxConfig();
+
+            // Verify we can load the service
+            config.AddService("PetStore", apiDoc);
+
+            // Ensure we use HTTPS protocol
+            Assert.Equal("https", apiDoc.GetScheme().Substring(0, 5));
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Connectors.Tests/BasicRestTests.cs:182]: >.Substring(0, 5) [](http://example.com/codeflow?start=52&length=16)

why Substring?
At least StartsWith() 
Or can we compare the whole string?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Pull request merged**